### PR TITLE
adds TTS generation status polling in Classic Editor

### LINKED ISSUE #1072: Add audio generation status check for Classic Editors
### Is your enhancement related to a problem? Please describe.

3.8.0 introduced Action Scheduler support for audio generation. In Classic Editors, when the audio generation is triggered, it displays the "Audio generation is in progress…" message, but it stays that way forever until the page is reloaded.

Image

This false negative results in users waiting for hours waiting on the generation to complete.

### Designs

_No response_

### Describe alternatives you've considered

Use Heartbeat-API to poll the results and update the message to something like "Audio generation complete"

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CacheMeOwside**
### Description of the Change

3.8.0 introduced Action Scheduler support for audio generation. The Classic Editor meta box renders the "in-progress" state when the audio generation is triggered so the "Audio generation is in progress…" message persists indefinitely with no client-side mechanism to detect completion.

This PR adds a 10-second polling. When the scheduled action completes, the in-progress text is replaced with the post-completion UI (checkboxes + audio player), without requiring a page reload.

Post-fix video (speed up 3x)

https://github.com/user-attachments/assets/03b62c90-4d46-4712-aafa-9326bd292726

Closes #1072

### How to test the Change

1. Enable Text to Speech with a configured provider; switch to Classic Editor.
2. Edit a post with content, tick **Enable audio generation**, click Update.
3. Confirm "Audio generation is in progress…" appears, and DevTools → Network shows `admin-ajax.php` POSTs every 10s with`action=classifai_get_tts_status`.                                                                                                                    
4. As soon as the TTS generation completes, the meta box swaps to the post-completion UI (two checkboxes + audio player) without a reload.

### Changelog Entry

 > Fixed - Classic Editor TTS meta box no longer hangs at "Audio generation is in progress…"; it now polls and swaps to the completed state without requiring a page reload.

### Credits

Props @username, ...

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: includes/Classifai/Features/TextToSpeech.php [hunk 1]
```diff
@@ -499,15 +511,66 @@ function_exists( 'as_has_scheduled_action' ) &&
...
 	}
 
+	/**
+	 * AJAX handler for the Classic Editor meta box's polling. Returns the
+	 * current audio generation status for the given post id.
+	 */
+	public function ajax_get_audio_generation_status() {
+		check_ajax_referer( 'classifai_tts_status', 'nonce' );
+
+		$post_id = isset( $_POST['post_id'] ) ? absint( $_POST['post_id'] ) : 0;
```
Comment (dkotter) [inline includes/Classifai/Features/TextToSpeech.php]:
```suggestion
		$post_id = isset( $_POST['post_id'] ) ? absint( wp_unslash( $_POST['post_id'] ) ) : 0;
```

File: src/js/admin.js [hunk 2]
```diff
@@ -445,6 +445,60 @@ document.addEventListener( 'DOMContentLoaded', function () {
 	} );
 } )( jQuery );
 
+// Poll admin-ajax to update the Classic Editor TTS meta box once audio generation completes.
```
Comment (dkotter) [inline src/js/admin.js]: Any reason to not put this code in the Text to Speech index.js file? That way it only loads when that feature is on?

**Review by dkotter — COMMENTED** [review]
Overall this looks good to me, left a few comments.

But seeing VIPCS and E2E tests failing. Not sure if that's related to this PR or not but ideally those all get cleaned up prior to us merging this

File: src/js/admin.js (hunk 2 above)
Comment (CacheMeOwside) [reply src/js/admin.js]: Thanks @dkotter. Moved the polling code into /text-to-speech/classic/index.js, enqueued only on Classic Editor when the TTS feature is enabled.

**Comment by CacheMeOwside** [comment]
@dkotter On the failing CI tests, looks like these are also failing on other recent PRs against develop, so they don't seem to be related to this PR. I'll dig into them tomorrow and try to figure out what's breaking here.

**Review by dkotter — APPROVED** [review]
Thanks for the PR! I've gotten tests running again and they are now passing here
